packages/google_sign_in/example/lib/main.dart

```diff
@@ -94,18 +97,45 @@ class SignInDemoState extends State<SignInDemo> {
     return null;
   }
 
-  Future<Null> _handleSignIn() async {
-    try {
-      await _googleSignIn.signIn();
-    } catch (error) {
-      print(error);
-    }
+  void _handleSignIn() {
+    _handleGoogleSignIn()
+        .then((GoogleSignInAccount user) => _afterSignIn)
+        .catchError((dynamic error) => _onSignInError);
```

Comment:
That's fine. I am curious, what if it were written differently? Such as using the test argument for checking if it's a PlatformException? 

```
handleSignIn()
    .then((FirebaseUser user) => _onSignIn(user))
    .catchError((e) => onSignInError(e), 
    test: (e) => e is PlatformException); // <<<---
```